src/eterna/ui/AnnotationDialog.ts

```diff
@@ -59,6 +59,23 @@ export default class AnnotationDialog extends Dialog<AnnotationData> {
 
     protected added(): void {
         super.added();
+        this.renderDialog(true);
+    }
+
+    public renderDialog(focusTitle: boolean): void {
+        const prevTitle = this._titleField?.input.text || '';
+        const prevBases = this._basesField?.input.text || '';
+
+        // Clear old objects
+        this.removeAllObjects();
+        this.container.children.forEach((child: DisplayObject) => {
+            child.destroy();
+        });
```

Comment:
As a note, this is generally not preferred, vs adding all objects in added(), and then only changing whatever it is you want to update (positions, etc). This is fine though since I really just want to get this out the door.

Reply:
The complication was that the dialog needed to be rerendered when updates occured — I'd opted for clearing out old child elements and replacing them with new ones based on the updated info. It appears though you're advocating for creating a completely new instance of the dialog upon updates?

Reply:
I'm advocating for changing already instantiated objects rather than destroying and recreating them

Reply:
As an example, there's a lot of layout methods throughout the code: https://github.com/eternagame/EternaJS/blob/dev/src/eterna/ui/EternaMenu.ts#L250

Reply:
Though note, there's probably too many of these, where instead we should be relying on `LayoutContainer`s

Reply:
Gotcha! Was unaware of the pattern. As a sidenote for future devs, it would be useful to record these sort of Eterna pattern best practices in some central place to avoid inconsistencies.

Suggestions on how to proceed forward? 1) Keep the same. 2) Adapt code to the pattern used in EternaMenu. 3) Use a different pattern you're hinting at that uses `LayoutContainer`.

Reply:
I'm inclined to say (1) just so we can get this out the door.

Reply:
If you're willing to do it though, I would prefer avoiding the "destroy everything and re-instantiate".

Reply:
I'll try to emulate the pattern (i.e. numero 2)